// @adminjs/themes/dist/themes/no-sidebar/components/TopBar.js
import { Box, CurrentUserNav } from '@adminjs/design-system';
import { ViewHelpers, useTranslation } from 'adminjs';
import React from 'react';
import { useSelector } from 'react-redux';
import { Navigation } from '../assets/Navigation.jsx';
const h = new ViewHelpers();
const TopBar = () => {
  const session = useSelector(state => state.session);
  const branding = useSelector(state => state.branding);
  const {
    logo,
    companyName
  } = branding;
  const {
    tb
  } = useTranslation();
  const dropActions = [{
    label: tb('logout'),
    href: h.logoutUrl(),
    icon: 'LogOut'
  }];
  return /*#__PURE__*/React.createElement(Box, {
    flex: true,
    flexDirection: "row",
    alignItems: "center",
    height: "navbarHeight",
    bg: "container"
  }, /*#__PURE__*/React.createElement(Box, {
    as: "a",
    href: h.dashboardUrl(),
    mx: "xl"
  }, logo ? /*#__PURE__*/React.createElement("img", {
    src: logo,
    alt: companyName
  }) : /*#__PURE__*/React.createElement("h1", null, companyName)), /*#__PURE__*/React.createElement(Navigation, null), /*#__PURE__*/React.createElement(Box, {
    as: "div",
    flex: true,
    flexGrow: 1
  }), session && /*#__PURE__*/React.createElement(CurrentUserNav, {
    name: session.email,
    title: session.title,
    avatarUrl: session.avatarUrl,
    dropActions: dropActions
  }));
};
export default TopBar;
